client/broadcast.go

```diff
@@ -80,6 +81,14 @@ func CheckTendermintError(err error, tx tmtypes.Tx) *sdk.TxResponse {
 	}
 }
 
+func checkTendermintErrorForBroadcast(err error, tx tmtypes.Tx, res *coretypes.ResultBroadcastTxCommit) *sdk.TxResponse {
+	if err == nil || strings.Contains(err.Error(), "transaction encountered error") {
```

Comment:
This logic seems weird. Could we e.g. put it in CheckTendermintError?

Reply:
>> we are using CheckTendermintError in 3 places https://github.com/cosmos/cosmos-sdk/blob/master/client/broadcast.go#L90-L137
> and in checkTendermintErrorForBroadcast I'm passing an extra arg res *coretypes.ResultBroadcastTxCommit
> the res *coretypes.ResultBroadcastTxCommit cannot be passed from the other 2 places, since it has different 
> res (which is *coretypes.ResultBroadcastTx)
> 
> I see
> Maybe for readability, we can simply do:
> if err == nil {
>   return sdk.NewResponseFormatBroadcastTxCommit(res)
> }
> // Add some comments here, e.g linke the Tendermint issue
> if strings.Contains(err.Error(), "transaction encountered error") {
>   return sdk.NewResponseFormatBroadcastTxCommit(res)
> }

updated it as you suggested